server/src/main/kotlin/suwayomi/tachidesk/server/ServerSetup.kt

```diff
@@ -161,25 +171,40 @@ fun applicationSetup() {
     LocalSource.register()
 
     // create system tray
-    if (serverConfig.systemTrayEnabled) {
+    serverConfig.subscribeTo(serverConfig.systemTrayEnabled, { systemTrayEnabled ->
         try {
-            systemTrayInstance
+            if (systemTrayEnabled) {
+                systemTrayInstance
+            } else {
+                systemTrayInstance?.remove()
+            }
         } catch (e: Throwable) { // cover both java.lang.Exception and java.lang.Error
             e.printStackTrace()
         }
-    }
+    }, ignoreInitialValue = false)
 
     // Disable jetty's logging
     System.setProperty("org.eclipse.jetty.util.log.announce", "false")
     System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StdErrLog")
     System.setProperty("org.eclipse.jetty.LEVEL", "OFF")
 
     // socks proxy settings
-    if (serverConfig.socksProxyEnabled) {
-        System.getProperties()["socksProxyHost"] = serverConfig.socksProxyHost
-        System.getProperties()["socksProxyPort"] = serverConfig.socksProxyPort
-        logger.info("Socks Proxy is enabled to ${serverConfig.socksProxyHost}:${serverConfig.socksProxyPort}")
-    }
+    serverConfig.subscribeTo(
+        combine(
+            serverConfig.socksProxyEnabled,
+            serverConfig.socksProxyHost,
+            serverConfig.socksProxyPort
+        ) { proxyEnabled, proxyHost, proxyPort ->
+            Triple(proxyEnabled, proxyHost, proxyPort)
+        },
+        { (proxyEnabled, proxyHost, proxyPort) ->
+            if (proxyEnabled) {
+                System.getProperties()["socksProxyHost"] = proxyHost
+                System.getProperties()["socksProxyPort"] = proxyPort
+                logger.info("Socks Proxy is enabled to $proxyHost:$proxyPort")
+            }
```

Comment:
Does this actually work?

Reply:
I forgot about that, I wasn't able to test that (and wasn't able to find where this is actually accessed), so some else would need to check if that is working

Reply:
yes, works